design/open/ttv-wizard-mvp.md

```diff
@@ -0,0 +1,271 @@
+# TimeToValue: Wizard MVP Design
+
+## Description
+
+This design document will specify a possible solution for the time-to-value [wizard MVP](https://github.com/treeverse/lakeFS/issues/3411).  
+The wizard will provide a quick and clear way to start interacting with lakeFS.
+It will do that by allowing the users to
+1. Initialize a new repository with a given namespace (and name) with a ‘main’ default branch.
+2. Import their data (write metadata) into the new repository’s main branch by specifying the data’s S3 bucket namespace.
+3. Get custom Spark configurations (and possibly custom Hive metastore configurations) to access lakeFS using the S3 Gateway (this is for the MVP).
+4. Summarise all actions performed (or skipped) in a README file which will be at the root of the initialized repository.
+
+---
+
+## System Overview
+
+![RouterFS with lakeFS URI](diagrams/Wizard.png)
+[(excalidraw file)](diagrams/wizard.excalidraw)
+
+### Wizard UI Component
+
+The wizard UI component is responsible for the user’s Spark onboarding process. The process is as follows:
+1. Create a repository (named as the user wishes) and a ‘main’ branch in it.
+2. Import the user’s data to ‘main’ and display a progress bar (which will show a link to required permissions). Only available on cloud-deployed lakeFS.
+3. Generate Spark configurations: the wizard will return Spark configurations for the users to use in their Spark core-site.xml file, [databricks key-value format](https://docs.databricks.com/clusters/configure.html#spark-configuration), or [EMR JSON](https://docs.aws.amazon.com/emr/latest/ReleaseGuide/emr-configure-apps.html) ([core-site](https://docs.aws.amazon.com/emr/latest/ReleaseGuide/emr-configure-apps.html)).
+   1. Asks the user to enter the lakeFS endpoint (by default it will be https://`location.host`)
+   2. Asks the user for the runtime environment of Spark: one of [Databricks, AWS EMR, spark-submit] ([Figma design](https://www.figma.com/file/haD9y599LzW6LvsYBI2xWU/Spark-use-case?node-id=31%3A200))
+4. Create a README file with all actions performed and commit it as a first commit to the ‘main’ branch of the created repository. If the Spark configuration step was performed, additional Metastore guidelines will be provided in the README.
+
+### Templating Service
+
+The [templating service](https://github.com/treeverse/lakeFS/pull/3373) is responsible for fetching, authenticating, and expanding the required templates and returning them to the client.  
+**Process**:  
+1. Get the S3 templates (locations should be specified in the incoming request). The files must be valid [`html/template`](https://pkg.go.dev/html/template) or [`text/template`](https://pkg.go.dev/text/template) parsable template texts.
```

Comment:
Not sure incoming request should entirely specify the location.  Would be happier if a set prefix were configured, and the incoming request could only select below that prefix.  Otherwise admins will have a very tough time ensuring the access is safe.

The proposed template service never allows access by arbitrary path precisely for this reason.  It protects access to templates on lakefs in some way; if you want to do the same but on S3, we will need to design similar protections for files from S3.